script.js

```diff
@@ -1,4 +1,294 @@
+      Your rolled dice result: <br><br>
+    dice 1 : ${player2dice1Arr} <br><br>
+    dice 2 : ${player2dice2Arr} <br><br>
+    Please choose the order of desired dice result. <br>
+    Type 1 and click enter if you wish for dice 1 to be the first number <br>
+    Type 2 and click enter if you wish for dice 2 to be the first number
+      `;
+
+    console.log(
+      `Invalid input, current mode should still be player choice : ${currentGameMode}`
+    );
+    return myOutputValue;
+  }
+  return myOutputValue;
+};
+
+var gameOutcome = function (player1, player2) {
+  myOutputValue = "";
+  if (player1Score > player2Score) {
+    myOutputValue = `Congratz ${player1} for winning!!! <br><br>
+    ${player1} BEATS ${player2} with a score of ${player1Score} against ${player2Score} <br><br>
+    Click enter to start another new game`;
+    return myOutputValue;
+  } else if (player2Score > player1Score) {
+    myOutputValue = `Congratz ${player2} for winning!!! <br><br>
+     ${player2} BEATS ${player1} with a score of ${player2Score} against ${player1Score}<br><br>
+    Click enter to start another new game`;
+    return myOutputValue;
+  }
+};
+
+//Beat That Game Function
+var beatThatGame = function () {};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var myOutputValue = "";
+
+  if (currentGameMode == "welcomeToBeatThat") {
+    //Opening message + Instructions
+    myOutputValue = `Welcome to BEAT THAT!<br><br>
+     BEFORE starting the game, please ENTER desired player 1 In Game Name (IGN) and click ENTER, followed by player 2. <br><br>
+      Game Rules: <br>
+      1. This is a 2 person, player vs player game. <br>
+2. The game will roll 2 dice and show the dice rolls result. For example, dice 1 rolled 3 and dice 2 rolled 4. <br>
+3. The player that rolled the dice will pick the order of the dice.(If they wanted the number 43, they would specify that the 2nd dice goes first.) <br>
+4. After both players have rolled and chosen their dice order, the player with the higher combined number wins.  `;
+
+    currentGameMode = "enterPlayer1";
+
+    return myOutputValue;
+  }
+  if (currentGameMode == "enterPlayer1") {
+    //Set player 1 IGN
+    player1 = input;
+
+    currentGameMode = "enterPlayer2";
+
+    console.log(`currentGameMode, ${currentGameMode}`);
+    console.log(`Player 1 IGN, player 1: ${player1}`);
+
+    myOutputValue = `Welcome ${player1}. Player 2, please enter your desired IGN and click enter. `;
+  } else if (currentGameMode == "enterPlayer2") {
+    //Set player 2 IGN
+    player2 = input;
+
+    currentGameMode = "player1Roll"; //Switch to Game Result for player 1
+
+    console.log(`currentGameMode, ${currentGameMode}`);
+    console.log(`Player 2 IGN, player 2: ${player2}`);
+
+    myOutputValue = `Welcome ${player2}. <br><br>
+    ${player1}, you may begin to start the game by clicking enter.`;
```

Comment:
this is ok but a bit repetitive, consider how you can refactor this so that there is less code repetition

one way i can think of is having a `playerNames` array instead of the `player1` and `player2` variables you currently have, then just have one game state to keep pushing new player names into the array until the desired amount of players is reached, upon which you switch the game state to 'player1Roll`